rs/state_manager/src/manifest/tests/computation.rs

```diff
@@ -329,6 +330,47 @@ fn test_simple_manifest_computation() {
     }
 }
 
+#[test]
+fn test_manifest_computation_skips_marker_file() {
+    let metrics_registry = MetricsRegistry::new();
+    let manifest_metrics = ManifestMetrics::new(&metrics_registry);
+    let dir = tempfile::TempDir::new().expect("failed to create a temporary directory");
+
+    let root = dir.path();
+    fs::write(root.join("root.bin"), vec![0u8; 1000]).expect("failed to create file 'test.bin'");
+    fs::write(root.join(UNVERIFIED_CHECKPOINT_MARKER), &[]).expect("failed to create marker file");
+
+    let subdir = root.join("subdir");
+    fs::create_dir_all(&subdir).expect("failed to create dir 'subdir'");
+    fs::write(subdir.join("memory"), vec![1u8; 2048]).expect("failed to create file 'memory'");
+    fs::write(subdir.join("queue"), vec![0u8; 0]).expect("failed to create file 'queue'");
+    fs::write(subdir.join("metadata"), vec![2u8; 1050]).expect("failed to create file 'queue'");
+
+    let mut thread_pool = scoped_threadpool::Pool::new(1);
+
+    // The debug_assertion in manifest computation panics in this test.
```

Comment:
I'm not sure if this is ok. We are making the assumption that tests only ever run in with debug assertions enabled, which I think is true atm, but doesn't have to be. I think we should only do this check if they are enabled.

Reply:
Good point. I use `cfg!(debug_assertions)` to distinguish debug and release modes in tests.